### ISSUE TITLE
The `CCDData` object finds non-image extensions when looking through a FITS file for images

### ISSUE THREAD

**Issue by mwcraig** [astropy/astropy#7595]
In `CCDData.read`, if the hdu isn't specified explicitly and the first HDU doesn't contain image data, then the code iterates through HDUs until it finds one with data....which is not necessarily image data. 

The offending code is at https://github.com/astropy/astropy/blob/master/astropy/nddata/ccddata.py#L503

**Comment by mwcraig** [comment]
To reproduce:

```python
from astropy.io import fits
from astropy.table import Table
from astropy.nddata import CCDData

import numpy as np

# Make a table with 10 rows, 20 columns 
tbl = Table(np.random.rand(10) for _ in range(20))

# A fancy 100x100 image
img = np.random.rand(100, 100)

# Make FITS file with table data followed by image data
hdul = fits.HDUList(hdus=[fits.PrimaryHDU(), fits.TableHDU(tbl.as_array()), fits.ImageHDU(img)])
hdul.writeto('image_last.fits')

# Read the fits file in
ccd = CCDData.read('image_last.fits', unit='adu')

# Expecting to get (100, 100), the size of the image
ccd.data.shape

# Instead, get the table, with shape (10,)
```

**Comment by mwcraig** [comment]
The fix I'm looking for here (in case someone else takes this up) is to check whether the HDU is an instance of `ImageHDU` rather than check that `datSpan` is nonzero. I'd also prefer iterating over the HDUs instead of iterating over an index. Invoking `len(hdus)` requires the whole file be scanned, which will be slow for large files.

**Comment by ricardogando** [comment]
Hi @mwcraig, thanks for the detailed description. Will go over this issue with a group of students @anaposses @nicolopes @rebecamelo

**Comment by mwcraig** [comment]
Excellent! Let me know if you need help with anything! Also, don't be shy about opening a pull request even if you think something isn't right or that it isn't done enough...it is easier to provide support and feedback on an opened PR 😀

**Comment by sunilk747** [comment]
Hi all. If nobody is working on this issue I would like to take this up.

**Comment by ricardogando** [comment]
Hi @sunilk747, @anaposses is preparing a PR for this issue. Do you mind taking another one? Thanks

**Comment by mwcraig** [comment]
@sunilk747 -- you might be interested in astropy/astropy#7526 or astropy/ccdproc#631

**Comment by pllim** [comment]
Fixed in #7739. Thanks, @anaposses !
